### ISSUE TITLE
Training Methodology Issue: Incorrect Dataset Mode

### ISSUE THREAD

**Issue by mojtaba-nafez** [zqhang/AnomalyCLIP#8]
I've encountered an issue in the paper's code regarding the training approach for the anomaly detector on the MVTec AD dataset. The problem lies in the train.py script at line 35 where the Dataset object for training is created without specifying mode="train":

`train_data = Dataset(root=args.train_data_path, transform=preprocess, target_transform=target_transform, dataset_name=args.dataset)`

This oversight leads to two critical problems:

1. The model mistakenly uses the test set for training.
2. The anomaly detection framework is deprived of abnormal data during training, yet it encounters anomalies during testing. This inconsistency suggests that the model might be functioning as a simple classifier rather than performing anomaly detection.

Could this be revised to ensure the correct dataset partitioning and training setup?

**Comment by cescomuch** [comment]
I have also encountered this behavior. How is this possible? It actually does not follow what is written in the paper.

**Comment by zqhang** [comment]
Hi, I believe this point has already been clarified in the paper. We fine-tune AnomalyCLIP using an auxiliary dataset and then test it on other datasets. Regarding the decision to set the test mode during training, this is because the MVTec dataset does not provide anomaly samples for fine-tuning AnomalyCLIP.
